### PR TITLE
Add timezone support to v3 SQL docs

### PR DESCRIPTION
**Pull request by sanderson**
- Updates information about time zones in SQL across all v3 products
- Updates SQL data types documentation
- Adds missing time and date functions
- Adds information about how timezones change the behavior of some functions.

---

- [x] Rebased/mergeable

### CODE REVIEW

File: content/influxdb/clustered/reference/sql/data-types.md [hunk 1]
```diff
@@ -122,10 +132,10 @@ Floats can be a decimal point, decimal integer, or decimal fraction.
 
 InfluxDB SQL supports the following DATE/TIME data types:
 
-| Name      | Data type | Description                                                          |
-| :-------- | :-------- | :------------------------------------------------------------------- |
-| TIMESTAMP | TIMESTAMP | TimeUnit::Nanosecond, None                                           |
-| INTERVAL  | INTERVAL  | Interval(IntervalUnit::YearMonth) or Interval(IntervalUnit::DayTime) |
+| SQL data type | Arrow data type                | Description                                  |
+| :------------ | :----------------------------- | :------------------------------------------- |
+| TIMESTAMP     | Timestamp(Nanosecond, None)    | Nanosecond timestamp with a time zone offset |
```
Comment (jeffreyssmith2nd) [inline content/influxdb/clustered/reference/sql/data-types.md:137]: Same comment as above

File: content/influxdb/cloud-serverless/reference/sql/operators/other.md [hunk 2]
```diff
@@ -44,20 +44,42 @@ SELECT 'Hello' || ' world' AS "Concatenated"
...
 
-| UTC with TZ offset       |
-| :----------------------- |
-| 2024-01-01T08:00:00.000Z |
+| Local time with TZ offset |
+| :------------------------ |
+| 2024-01-01T00:00:00-08:00 |
+
+{{% /expand %}}
+{{< /expand-wrapper >}}
```
Comment (jeffreyssmith2nd) [inline content/influxdb/cloud-serverless/reference/sql/operators/other.md:85]: Same comment as below

File: content/influxdb/cloud-serverless/reference/sql/data-types.md [hunk 3]
```diff
@@ -122,10 +132,10 @@ Floats can be a decimal point, decimal integer, or decimal fraction.
 
 InfluxDB SQL supports the following DATE/TIME data types:
 
-| Name      | Data type | Description                                                          |
-| :-------- | :-------- | :------------------------------------------------------------------- |
-| TIMESTAMP | TIMESTAMP | TimeUnit::Nanosecond, None                                           |
-| INTERVAL  | INTERVAL  | Interval(IntervalUnit::YearMonth) or Interval(IntervalUnit::DayTime) |
+| SQL data type | Arrow data type                | Description                                  |
+| :------------ | :----------------------------- | :------------------------------------------- |
+| TIMESTAMP     | Timestamp(Nanosecond, None)    | Nanosecond timestamp with a time zone offset |
```
Comment (jeffreyssmith2nd) [inline content/influxdb/cloud-serverless/reference/sql/data-types.md:137]: Same as above

File: content/influxdb/cloud-dedicated/reference/sql/operators/other.md [hunk 4]
```diff
@@ -45,20 +45,42 @@ SELECT 'Hello' || ' world' AS "Concatenated"
...
 
-| UTC with TZ offset       |
-| :----------------------- |
-| 2024-01-01T08:00:00.000Z |
+| Local time with TZ offset |
+| :------------------------ |
+| 2024-01-01T00:00:00-08:00 |
+
+{{% /expand %}}
+{{< /expand-wrapper >}}
```
Comment (jeffreyssmith2nd) [inline content/influxdb/cloud-dedicated/reference/sql/operators/other.md:86]: Same as below

File: content/influxdb/cloud-dedicated/reference/sql/data-types.md [hunk 5]
```diff
@@ -13,16 +13,26 @@ related:
...
 
+## SQL and Arrow data types
+
+In SQL, each column, expression, and parameter has a data type.
+A data type is an attribute that specifies the type of data that the object can hold.
 DataFusion uses the [Arrow](https://arrow.apache.org/) type system for query execution.
-Data types stored in InfluxDB's storage engine are mapped to SQL data types at query time.
+All SQL types are mapped to [Arrow data types](https://docs.rs/arrow/latest/arrow/datatypes/enum.DataType.html).
+
+Both SQL and Arrow data types play an important role is how data is operated on
```
Comment (jstirnaman) [inline content/influxdb/cloud-dedicated/reference/sql/data-types.md:29]:
```suggestion
Both SQL and Arrow data types play an important role in how data is operated on
```

File: content/influxdb/cloud-dedicated/reference/sql/functions/time-and-date.md [hunk 6]
```diff
@@ -22,20 +22,26 @@ InfluxDB's SQL implementation supports time and date functions that are useful w
...
 
 Returns the current UTC date.
 
 {{% note %}}
 `current_date` returns a `DATE32` Arrow type, which isn't supported by InfluxDB.
-To use with InfluxDB, [cast the return value to a timestamp](/influxdb/cloud-dedicated/query-data/sql/cast-types/#cast-to-a-timestamp-type).
+To use with InfluxDB, [cast the return value to a timestamp or string](/influxdb/cloud-dedicated/query-data/sql/cast-types/).
 {{% /note %}}
 
 The `current_date()` return value is determined at query time and will return
```
Comment (jstirnaman) [inline content/influxdb/cloud-dedicated/reference/sql/functions/time-and-date.md:47]:
```suggestion
The `current_date()` return value is determined at query time and returns
```

File: content/influxdb/cloud-dedicated/reference/sql/functions/time-and-date.md [hunk 7]
```diff
@@ -121,8 +127,7 @@ Calculates time intervals and returns the start of the interval nearest to the s
 Use `date_bin` to downsample time series data by grouping rows into time-based "bins" or "windows"
 and applying an aggregate or selector function to each window.
 
-For example, given an input
-timestamp of `2023-01-01T18:18:18Z`, if you bin data into 15-minute intervals, the output timestamp is `2023-01-01T18:15:00Z`--the start of the 15-minute bin that includes the input.
+For example, if you "bin" or "window" data into 15 minute intervals, an input timestamp of `2023-01-01T18:18:18Z` will be updated to the start time of the 15 minute bin it is in: `2023-01-01T18:15:00Z`.
```
Comment (jstirnaman) [inline content/influxdb/cloud-dedicated/reference/sql/functions/time-and-date.md:130]:
```suggestion
For example, if you "bin" or "window" data into 15-minute intervals, an input timestamp of `2023-01-01T18:18:18Z` will be updated to the start time of the 15-minute bin it is in: `2023-01-01T18:15:00Z`.
```

File: content/influxdb/cloud-dedicated/reference/sql/functions/time-and-date.md [hunk 8]
```diff
@@ -608,6 +653,156 @@ WHERE
...
+| Mon  1-Jan-2024 12:22:01 |
+
+{{% /expand %}}
+{{< /expand-wrapper >}}
+
+## to_date
+
+Converts a value to a date (`YYYY-MM-DD`).
+Supports strings and numeric types as input.
+Strings are parsed as `YYYY-MM-DD` unless another format is specified. if no Chrono formats are provided.
```
Comment (jstirnaman) [inline content/influxdb/cloud-dedicated/reference/sql/functions/time-and-date.md:700]: Incomplete sentence.

File: content/influxdb/cloud-dedicated/reference/sql/functions/time-and-date.md [hunk 9]
```diff
@@ -825,3 +1020,55 @@ SELECT to_timestamp_seconds('01:01:59.123456789 01-01-2024', '%c', '%+', '%H:%M:
 
 {{% /expand %}}
 {{< /expand-wrapper >}}
+
+## to_unixtime
+
+Converts a value to seconds since the [Unix epoch](/influxdb/cloud-dedicated/reference/glossary/#unix-epoch).
+Supports strings, timestamps, and floats as input.
+Strings are parsed as RFC3339nano timestamps if no
```
Comment (jstirnaman) [inline content/influxdb/cloud-dedicated/reference/sql/functions/time-and-date.md:1028]:
```suggestion
Strings are parsed as [RFC3339Nano timestamps](/influxdb/cloud-dedicated/reference/glossary/#rfc3339nano-timestamp) if no
```

File: content/influxdb/cloud-serverless/reference/sql/functions/time-and-date.md [hunk 10]
```diff
@@ -22,19 +22,26 @@ InfluxDB's SQL implementation supports time and date functions that are useful w
...
 
 Returns the current UTC date.
 
 {{% note %}}
 `current_date` returns a `DATE32` Arrow type, which isn't supported by InfluxDB.
-To use with InfluxDB, [cast the return value to a timestamp](/influxdb/cloud-serverless/query-data/sql/cast-types/#cast-to-a-timestamp-type).
+To use with InfluxDB, [cast the return value to a timestamp or string](/influxdb/cloud-serverless/query-data/sql/cast-types/).
 {{% /note %}}
 
 The `current_date()` return value is determined at query time and will return
```
Comment (jstirnaman) [inline content/influxdb/cloud-serverless/reference/sql/functions/time-and-date.md:47]:
```suggestion
The `current_date()` return value is determined at query time and returns
```

File: content/influxdb/cloud-serverless/reference/sql/functions/time-and-date.md [hunk 11]
```diff
@@ -610,6 +653,156 @@ WHERE
...
+| Mon  1-Jan-2024 12:22:01 |
+
+{{% /expand %}}
+{{< /expand-wrapper >}}
+
+## to_date
+
+Converts a value to a date (`YYYY-MM-DD`).
+Supports strings and numeric types as input.
+Strings are parsed as `YYYY-MM-DD` unless another format is specified. if no Chrono formats are provided.
```
Comment (jstirnaman) [inline content/influxdb/cloud-serverless/reference/sql/functions/time-and-date.md:700]: Incomplete sentence

File: content/influxdb/cloud-serverless/reference/sql/functions/time-and-date.md [hunk 12]
```diff
@@ -827,3 +1020,55 @@ SELECT to_timestamp_seconds('01:01:59.123456789 01-01-2024', '%c', '%+', '%H:%M:
 
 {{% /expand %}}
 {{< /expand-wrapper >}}
+
+## to_unixtime
+
+Converts a value to seconds since the [Unix epoch](/influxdb/cloud-serverless/reference/glossary/#unix-epoch).
+Supports strings, timestamps, and floats as input.
+Strings are parsed as RFC3339nano timestamps if no
```
Comment (jstirnaman) [inline content/influxdb/cloud-serverless/reference/sql/functions/time-and-date.md:1028]:
```suggestion
Strings are parsed as [RFC3339Nano timestamps](/influxdb/cloud-serverless/reference/glossary/#rfc3339nano-timestamp) if no
```

File: content/influxdb/clustered/reference/sql/data-types.md [hunk 13]
```diff
@@ -13,16 +13,26 @@ related:
...
 
+## SQL and Arrow data types
+
+In SQL, each column, expression, and parameter has a data type.
+A data type is an attribute that specifies the type of data that the object can hold.
 DataFusion uses the [Arrow](https://arrow.apache.org/) type system for query execution.
-Data types stored in InfluxDB's storage engine are mapped to SQL data types at query time.
+All SQL types are mapped to [Arrow data types](https://docs.rs/arrow/latest/arrow/datatypes/enum.DataType.html).
+
+Both SQL and Arrow data types play an important role is how data is operated on
```
Comment (jstirnaman) [inline content/influxdb/clustered/reference/sql/data-types.md:29]:
```suggestion
Both SQL and Arrow data types play an important role in how data is operated on
```

File: content/influxdb/clustered/reference/sql/functions/time-and-date.md [hunk 14]
```diff
@@ -22,19 +22,26 @@ InfluxDB's SQL implementation supports time and date functions that are useful w
...
 
 Returns the current UTC date.
 
 {{% note %}}
 `current_date` returns a `DATE32` Arrow type, which isn't supported by InfluxDB.
-To use with InfluxDB, [cast the return value to a timestamp](/influxdb/clustered/query-data/sql/cast-types/#cast-to-a-timestamp-type).
+To use with InfluxDB, [cast the return value to a timestamp or string](/influxdb/clustered/query-data/sql/cast-types/).
 {{% /note %}}
 
 The `current_date()` return value is determined at query time and will return
```
Comment (jstirnaman) [inline content/influxdb/clustered/reference/sql/functions/time-and-date.md:47]:
```suggestion
The `current_date()` return value is determined at query time and returns
```

File: content/influxdb/clustered/reference/sql/functions/time-and-date.md [hunk 15]
```diff
@@ -823,3 +1020,55 @@ SELECT to_timestamp_seconds('01:01:59.123456789 01-01-2024', '%c', '%+', '%H:%M:
 
 {{% /expand %}}
 {{< /expand-wrapper >}}
+
+## to_unixtime
+
+Converts a value to seconds since the [Unix epoch](/influxdb/clustered/reference/glossary/#unix-epoch).
+Supports strings, timestamps, and floats as input.
+Strings are parsed as RFC3339nano timestamps if no
```
Comment (jstirnaman) [inline content/influxdb/clustered/reference/sql/functions/time-and-date.md:1028]:
```suggestion
Strings are parsed as [RFC3339Nano timestamps](/influxdb/clustered/reference/glossary/#rfc3339nano-timestamp) if no
```

File: content/influxdb/clustered/reference/sql/functions/time-and-date.md [hunk 16]
```diff
@@ -606,6 +653,156 @@ WHERE
...
+| Mon  1-Jan-2024 12:22:01 |
+
+{{% /expand %}}
+{{< /expand-wrapper >}}
+
+## to_date
+
+Converts a value to a date (`YYYY-MM-DD`).
+Supports strings and numeric types as input.
+Strings are parsed as `YYYY-MM-DD` unless another format is specified. if no Chrono formats are provided.
```
Comment (jstirnaman) [inline content/influxdb/clustered/reference/sql/functions/time-and-date.md:700]: Incomplete sentence

**Review by jstirnaman — CHANGES_REQUESTED** [review]

**Review by jstirnaman — APPROVED** [review]
LGTM. Maybe we need a note about the potential bug Jeff pointed out?

**Comment by sanderson** [comment]
@jeffreyssmith2nd I know this is an upstream bug. Are we going to hold off pushing the TZ support out until this bug is fixed? When should these doc changes go live?

**Comment by jeffreyssmith2nd** [comment]
> @jeffreyssmith2nd I know this is an upstream bug. Are we going to hold off pushing the TZ support out until this bug is fixed? When should these doc changes go live?

I don't know yet and am headed out for a long weekend. I'll check in with the team Tuesday on if we should delay or not.

Until then, I think we should hold the docs changes.

File: content/influxdb/cloud-serverless/reference/sql/functions/time-and-date.md (hunk 11 above)
Comment (jstirnaman) [reply content/influxdb/cloud-serverless/reference/sql/functions/time-and-date.md]: > Incomplete sentence

> Strings are parsed as `YYYY-MM-DD` unless another format is specified. if no Chrono formats are provided.

?

**Review by jstirnaman — CHANGES_REQUESTED** [review]

**Comment by sanderson** [comment]
@jstirnaman I think you may have been viewing the diff of an old commit. I went through and fixed all of those.

**Review by jstirnaman — APPROVED** [review]

**Comment by jeffreyssmith2nd** [comment]
@sanderson Still on hold for the docs changes, waiting on some upstream discussion on if the `::timestamp` thing is actually a bug or if it is intended behavior. 

In the meantime, I'm removing all references to it in the docs where it isn't necessary to prevent any confusion for people down the line.

I will update the blog post and make a corresponding PR here as well once we know for sure what behavior we want out of `::timestamp`

**Comment by jeffreyssmith2nd** [comment]
@sanderson This can be rolled out whenever you are ready, thanks for your help and patience with this one!

**Pull request merged**